core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/geodistance/GeoDistanceParser.java

```diff
@@ -164,6 +162,11 @@ public AggregatorFactory parse(String aggregationName, XContentParser parser, Se
         return new GeoDistanceFactory(aggregationName, vsParser.config(), InternalGeoDistance.FACTORY, origin, unit, distanceType, ranges, keyed);
     }
 
+    private static String getSearchParseExceptionMessage(String currentFieldName, String aggregationName) {
+        return "Failed when parsing field: '" +  currentFieldName +
```

Comment:
Actually the main confusion in the original issue was that the error that was thrown did not say which field in the request caused the error. However with this fix as it stands, if the user sends the wrong type of object for a field (say a String instead of an object) the error will not tell them what is wrong. I think we should add the token to the message, and possibly also the token we expected in this case too.